enzyme/tools/enzyme-tblgen/blasDiffUseUpdater.h

```diff
@@ -49,37 +49,20 @@ void emit_BLASDiffUse(TGPattern &pattern, llvm::raw_ostream &os) {
   }
 
   emit_scalar_caching(pattern, os);
-  // we currently cache all vecs before we cache all matrices
-  // once fixed we can merge this calls
   for (size_t i = 0; i < nameVec.size(); i++) {
     auto ty = typeMap.lookup(i);
-    if (ty != ArgType::vincData)
+    if (ty != ArgType::vincData && ty != ArgType::mldData)
       continue;
-    assert(typeMap.lookup(i + 1) == ArgType::vincInc);
-    emit_mat_vec_caching(pattern, i, os);
-  }
-  for (size_t i = 0; i < nameVec.size(); i++) {
-    auto ty = typeMap.lookup(i);
-    if (ty != ArgType::mldData)
-      continue;
-    assert(typeMap.lookup(i + 1) == ArgType::mldLD);
     emit_mat_vec_caching(pattern, i, os);
   }
 
   for (size_t argPos = (lv23 ? 1 : 0); argPos < typeMap.size(); argPos++) {
     auto users = argUsers.lookup(argPos);
     auto name = nameVec[argPos];
     size_t i = (lv23 ? argPos - 1 : argPos);
-    os << "  if (val == arg_" << name << " && !cache_" << name << ") {\n";
-    for (auto a : users) {
-      auto name = nameVec[a];
-      // The following shows that I probably should change the tblgen
-      // logic and the Blas.td declaration
-      if (a == i) // a == i? argpos ?
-        continue;
-      os << "    if (active_" << name << ") return true;\n";
-    }
-    os << "  }\n";
+    os << "  if (val == arg_" << name << " && need_" << name << " && !cache_"
```

Comment:
Like mentioned on slack, I don't think it's right to do cache checks here since this analysis is used by cache analysis.

Can you revert the cache change?

Reply:
Ah sorry I meant on here, not slack.

But yeah I believe there's an effective circular dependency, which here will be manifested by inconsistent results (not a compile time error), which is potentially very bad.

Reply:
just to clarify @wsmoses. If it existend then it existed just on the same level before. I did use exactly the same logic before this change, I just re-did some of the calculation. If you want this to be fundamentally change reverting will not change it, we would rather need to see what to use else